src/pip/_internal/models/direct_url.py

```diff
@@ -205,20 +206,25 @@ def from_dict(cls, d: Dict[str, Any]) -> "DirectUrl":
             ),
         )
 
-    def to_dict(self) -> Dict[str, Any]:
+    def to_dict(self, *, keep_hash_key: bool = True) -> Dict[str, Any]:
```

Comment:
```suggestion
    def to_dict(self, *, keep_legacy_hash_key: bool = True) -> Dict[str, Any]:
```

A little nitpick. Otherwise this PR generally looks good to me.